datasource/src/main/scala/quasar/plugin/googlebigtable/datasource/Evaluator.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.googlebigtable.datasource
+
+import slamdata.Predef._
+
+import quasar.common.data.{CLong, CString, RObject, RValue}
+
+import java.lang.Math
+import scala.collection.JavaConverters._
+
+import cats.effect.{ConcurrentEffect, Sync}
+
+import com.google.api.gax.rpc.StreamController
+import com.google.api.gax.rpc.StateCheckingResponseObserver
+import com.google.cloud.bigtable.data.v2.BigtableDataClient
+import com.google.cloud.bigtable.data.v2.models.{Query => GQuery, Row}
+
+import fs2.Stream
+
+class Evaluator[F[_]: ConcurrentEffect](client: BigtableDataClient, query: GQuery, maxQueueSize: Int) {
+  import Evaluator._
+
+  def evaluate(): Stream[F, RValue] = {
+    val handler = Observer.handler[F](client.readRowsAsync(query, _))
+    CallbackHandler.toStream[F, Row](handler, maxQueueSize).map(toRValue(_))
+  }
+}
+
+object Evaluator {
+
+  val DefaultMaxQueueSize = 10
+
+  def apply[F[_]: ConcurrentEffect](client: BigtableDataClient, query: GQuery, maxQueueSize: Int): Evaluator[F] =
+    new Evaluator[F](client, query, maxQueueSize)
+
+  def toRValue(row: Row): RValue = {
```

Comment:
If possible, you might consider defining `QDataDecode[Row]` to avoid the conversion to `RValue`.

Reply:
I did consider `QDataDecode[Row]`, but went for `RValue` (choosing developer convenience over performance)